### PR TITLE
[VarDumper] Towards PHP7 support

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

That's a start

### CODE REVIEW

File: src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php [hunk 1]
```diff
@@ -69,23 +70,23 @@ public static function castErrorException(\ErrorException $e, array $a, Stub $st
 
     public static function castThrowingCasterException(ThrowingCasterException $e, array $a, Stub $stub, $isNested)
     {
-        $b = (array) $a["\0Exception\0previous"];
+        $b = (array) $a[$xPrefix.'previous'];
```
Comment (staabm) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: Undefined var in this scope

Comment (nicolas-grekas) [inline src/Symfony/Component/VarDumper/Caster/ExceptionCaster.php]: indeed, fixed, thanks

File: src/Symfony/Component/VarDumper/Tests/HtmlDumperTest.php [hunk 2]
```diff
@@ -81,7 +80,7 @@ public function testGet()
   </samp>}
   "<span class=sf-dump-key>closure</span>" => <span class=sf-dump-note>Closure</span> {<a class=sf-dump-ref>#%d</a><samp>
     <span class=sf-dump-meta>reflection</span>: """
-      <span class=sf-dump-str title="%d characters">Closure [ &lt;user&gt; {$closureLabel} Symfony\Component\VarDumper\Tests\Fixture\{closure} ] {</span>
+      <span class=sf-dump-str title="%d characters">Closure [ &lt;user&gt; %s
```
Comment (stof) [inline src/Symfony/Component/VarDumper/Tests/HtmlDumperTest.php]: you should keep the closing space as part of the assertion

**Comment by nicolas-grekas** [comment]
This PR is ready. The last failing test should be fixed once https://bugs.php.net/69376 is.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**